core/domain/question_services_test.py

```diff
@@ -31,90 +32,120 @@ def setUp(self):
+        self.assertEqual(model.language_code, self.language_code)
+        self.assertEqual(model.version, 2)
 
     def test_delete_question(self):
-        question = question_domain.Question(
-            'dummy', self._create_valid_question_data('ABC'), 1, 'en')
-
-        question_id = question_services.add_question(self.owner_id, question)
-        question_services.delete_question(
-            self.owner_id, question_id)
+        question_services.delete_question(self.owner_id, self.question_id)
 
         with self.assertRaisesRegexp(Exception, (
             'Entity for class QuestionModel with id %s not found' % (
-                question_id))):
-            question_models.QuestionModel.get(question_id)
+                self.question_id))):
+            question_models.QuestionModel.get(self.question_id)
 
     def test_update_question(self):
-        question_data = self._create_valid_question_data('ABC')
-        question_id = 'dummy'
-        question_data_schema_version = 1
-        language_code = 'en'
-        question = question_domain.Question(
-            question_id, question_data, question_data_schema_version,
-            language_code)
-
         new_question_data = self._create_valid_question_data('DEF')
-        question_id = question_services.add_question(self.owner_id, question)
         change_dict = {'cmd': 'update_question_property',
-                       'property_name': 'question_data',
+                       'property_name': 'question_state_data',
                        'new_value': new_question_data,
-                       'old_value': question_data}
+                       'old_value': self.question_state_data}
         change_list = [question_domain.QuestionChange(change_dict)]
 
         question_services.update_question(
-            self.owner_id, question_id, change_list, (
-                'updated question data'))
+            self.owner_id, self.question_id, change_list,
+            ('updated question data'))
+
+        model = question_models.QuestionModel.get(self.question_id)
+        self.assertEqual(model.question_state_data, new_question_data)
+        self.assertEqual(model.version, 4)
+        self.assertEqual(
+            model.language_code, self.language_code)
+
+    def test_compute_summary_of_question(self):
+        html_content = '<b>This is updated text.</b>'
+        question_summary = question_services.compute_summary_of_question(
+            self.question, self.owner_id)
+        question_summary.update_html_content(html_content)
 
-        model = question_models.QuestionModel.get(question_id)
-        self.assertEqual(model.question_data, new_question_data)
+        self.assertEqual(question_summary.id, 'dummy')
         self.assertEqual(
-            model.question_data_schema_version, question_data_schema_version)
-        self.assertEqual(model.language_code, language_code)
+            question_summary.question_content,
+            html_cleaner.clean(html_content))
```

Comment:
Don't import html_cleaner. Use the string directly. Write the cleaned string explicitly if needed (that's actually a good test to make sure sanitization happens).